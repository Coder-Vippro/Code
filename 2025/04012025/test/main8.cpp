#include <bits/stdc++.h>
using namespace std;
bool tcs(int n)
{
    int tong=0;
    while(n>0)
    {
        tong+=n%10;
        n/=10;
    }
    return tong%2==0;
}
int dem=0;
int main()
{
    //freopen("main8.inp","r",stdin);
    //freopen("main8.out","w",stdout);
    int n;
    cin>>n;
    if(n%2==0 && tcs(n))
    {
        cout<<n/2;
    }
    else if(n%2==1 && tcs(n))
    {
        cout<<n/2;
    }
    else if(n%2==0 && !tcs(n))
    {
        cout<<n/2-1;
    }
    else cout<<n/2;
    /*while(n--)
    {
        int t;
        cin>>t;
        dem=0;
        for(int i=1;i<=t;i++)
        {
            if(tcs(i))
            {
                dem++;
            }
        }
        cout<<dem<<'\n';
    }
    */
}