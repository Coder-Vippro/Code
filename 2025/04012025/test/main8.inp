7
100
200
312
321
222
111
731