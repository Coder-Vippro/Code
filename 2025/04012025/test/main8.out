49
100
156
160
111
55
365
